boot/efi/multiboot.c

```diff
@@ -144,6 +151,11 @@ MultibootSaveMemoryMap(UINTN *mapkey)
    do {
 
       UINT32 type = EfiToMultibootMemType(desc->Type);
+
+      if (desc->Attribute & EFI_MEMORY_RUNTIME) {
```

Comment:
That's a dirty hack. If the `desc->Type` is not enough for `EfiToMultibootMemType()` to work properly, then pass to it also the `desc->Attribute` and determine the type correctly there.